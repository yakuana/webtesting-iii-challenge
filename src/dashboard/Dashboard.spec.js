// Test away

// react imports 
import React from "react";
import renderer from "react-test-renderer"; 
import { render, fireEvent } from "@testing-library/react";

// components 
import DashBoard from "./Dashboard.js"; 

describe("<DashBoard />", () => {
    it("shows the controls and display", () => { 

        // generates a DOM tree
        const dashTree = renderer.create(<DashBoard />); 

        // snapshots are a JSON representation of the DOM tree
        expect(dashTree.toJSON()).toMatchSnapshot();
    })
})