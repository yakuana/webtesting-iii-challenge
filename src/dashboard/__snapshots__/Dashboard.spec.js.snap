// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashBoard /> shows the controls and display 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="display-un-lock"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="display-opn-close"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="un-lock-gate-btn"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="open-close-gate-btn"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
